### ISSUE TITLE
Critical bug in uf_addmarginal()

### ISSUE THREAD

**Issue by olafdimigen** [unfoldtoolbox/unfold#67]
Function **uf_addmarginal** produces error with the attached test dataset.

Most likely because either of these reasons: 
- in the model, a rather large number of event types (e.g. {S 11}, {S 12}, {S 13},...) is grouped, e.g. for the intercept 
- more likely: model contains interaction term between categorical variables

Unfortunately, the code at the end of uf_addmarginal is quite obfuscated. Script and data to reproduce bug are found here:
[https://www.dropbox.com/s/6cmy9t4693rzt4o/bugreport_addmarginal.zip?dl=0](https://www.dropbox.com/s/6cmy9t4693rzt4o/bugreport_addmarginal.zip?dl=0)

The ZIP file contains:
- bugreport_addmarginal.mat  = test data containing  "condensed" model output
- github_bugreport_addmarginal = test script which reproduces the error
- uf_addmarginal_bugreport = version of uf_addmarginal() with improved comments and the lines producing the bug highlighted

**Comment by behinger** [comment]
One design decision I regret is to keep multiple events as cell arrays instead of combining them in some way... now it bites me. This is due to multiple events types.

I fixed it now, it is a bit opaque because one has to cycle through the cell arrays manually. Thanks for the comments, I added some of my own to make the last part more obvious whats going on.

There is one thing that worries me: I had ``` 
        if length(e_Idx) == 1
            continue 
        end
```
Included, but I really don't know anymore why. It intuitively does not make any sense to me. I will write a unittest and check it out in more detail
